src/Sulu/Bundle/CustomUrlBundle/Request/CustomUrlRequestProcessor.php

```diff
@@ -162,6 +162,14 @@ private function matchCustomUrl($url, PortalInformation $portalInformation, Requ
             $this->environment
         );
 
+        if (0 === count($portalInformations)) {
+            return $this->processPortalInformation(
+                $request,
+                $portalInformation,
+                ['customUrlRoute' => $routeDocument, 'customUrl' => $customUrlDocument]
+            );
+        }
```

Comment:
it tells the custom-url route-provider that there is a custom-url for given URL. Then the provider can check if it is valid or not.